### ISSUE TITLE
Batch normalization in the TFGAN library

### ISSUE THREAD

**Issue by kinsumliu** [tensorflow/models#3287]
### System information
- **What is the top-level directory of the model you are using**: TFGAN
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**: 1.5
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
@joel-shor 
In the TFGAN (https://github.com/tensorflow/models/blob/master/research/gan/tutorial.ipynb), the network uses batch normalization layer. According to the original paper, the behavior of batch norm is different in training and inference. The mean and variance of the normalization should not be updated during inference. Instead, inference just use the fixed statistics collected from training. But now, the implementation of TFGAN does not follow that.

### Source code / logs
For example in the tutorial.ipynb, the evaluation is done by this:

```
num_images_to_eval = 500
MNIST_CLASSIFIER_FROZEN_GRAPH = './mnist/data/classify_mnist_graph_def.pb'

# For variables to load, use the same variable scope as in the train job.
with tf.variable_scope('Generator', reuse=True):
    eval_images = gan_model.generator_fn(
        tf.random_normal([num_images_to_eval, noise_dims]))

eval_score = util.mnist_score(eval_images, MNIST_CLASSIFIER_FROZEN_GRAPH)
```

Although it reuses the scope of the generator, in the `generator_fn`,

```
  with tf.contrib.framework.arg_scope(
      [layers.fully_connected, layers.conv2d_transpose],
      activation_fn=tf.nn.relu, normalizer_fn=layers.batch_norm,
      weights_regularizer=layers.l2_regularizer(weight_decay)):
```

The correct way of using this for evaluation should be:

```
  with tf.contrib.framework.arg_scope(
      [layers.fully_connected, layers.conv2d_transpose],
      activation_fn=tf.nn.relu, normalizer_fn=layers.batch_norm,
      normalizer_params={'is_training': False},
      weights_regularizer=layers.l2_regularizer(weight_decay)):
```

As stated in `tf.contrib.layers.batch_norm`,

is_training: Whether or not the layer is in training mode. In training mode it would accumulate the statistics of the moments into moving_mean and moving_variance using an exponential moving average with the given decay. When it is not in training mode then it would use the values of the moving_mean and the moving_variance.

**Comment by joel-shor** [comment]
Good catch. PR #3370 fixes this.
